File: src/main/java/com/sixtythreeebays/service/ReturnService.java
package com.sixtythreeebays.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.sixtythreeebays.domain.Return;
import com.sixtythreeebays.dto.ReturnDTO;
import com.sixtythreeebays.dto.ReturnSearchDTO;
import com.sixtythreeebays.dto.ReturnPageDTO;
import com.sixtythreeebays.dto.ReturnConvertCriteriaDTO;
import com.sixtythreeebays.service.GenericService;
import com.sixtythreeebays.dto.common.RequestDTO;
import com.sixtythreeebays.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface ReturnService extends GenericService<Return, Integer> {

	List<Return> findAll();

	ResultDTO addReturn(ReturnDTO returnDTO, RequestDTO requestDTO);

	ResultDTO updateReturn(ReturnDTO returnDTO, RequestDTO requestDTO);

    Page<Return> getAllReturns(Pageable pageable);

    Page<Return> getAllReturns(Specification<Return> spec, Pageable pageable);

	ResponseEntity<ReturnPageDTO> getReturns(ReturnSearchDTO returnSearchDTO);
	
	List<ReturnDTO> convertReturnsToReturnDTOs(List<Return> returns, ReturnConvertCriteriaDTO convertCriteria);

	ReturnDTO getReturnDTOById(Integer returnId);







}

File: sixtythreeebays_ui/src/services/ReturnService.js
import http from "../http-common"; 

class ReturnService {
  getAllReturns(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/return/returns`, searchDTO);
  }

  get(returnId) {
    return this.getRequest(`/return/${returnId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/return?field=${matchData}`, null);
  }

  addReturn(data) {
    return http.post("/return/addReturn", data);
  }

  update(data) {
  	return http.post("/return/updateReturn", data);
  }
  
  uploadImage(data,returnId) {
  	return http.postForm("/return/uploadImage/"+returnId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new ReturnService();
